PhysicsTools/HepMCCandAlgos/plugins/ModelFilter.cc

```diff
@@ -1,11 +1,46 @@
-#include "PhysicsTools/HepMCCandAlgos/interface/ModelFilter.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/global/EDFilter.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "SimDataFormats/GeneratorProducts/interface/LHEEventProduct.h"
+
 #include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
 
 #include <iostream>
 #include <memory>
 #include <sstream>
 #include <cstdlib>
 
+/**
+    The ModelFilter class will select events in a "soup" MC
+    (like the SUSY signal MC) from the comments of LHEEventProduct
+    that match "modelTag". The user can require the value of that
+    parameter to lie between a min and max value.
+ */
+
+namespace edm {
+
+  class ModelFilter : public edm::global::EDFilter<> {
+  public:
+    explicit ModelFilter(const edm::ParameterSet&);
+
+    static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+    typedef std::vector<std::string>::const_iterator comments_const_iterator;
+
+  private:
+    bool filter(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
+    static std::vector<std::string> split(std::string const& fstring, std::string const& splitter);
+
+    edm::EDGetTokenT<LHEEventProduct> tokenSource_;
```

Comment:
All these parameters could be made `const`, and the same applies to several other classes touched by this PR.
I imagine that all these kind of (non mandatory) optimizations can be defereed to another PR, if wanted.